src/main/java/com/thealgorithms/others/SkylineProblem.java

```diff
@@ -4,34 +4,61 @@
 import java.util.Iterator;
 import java.util.Scanner;
 
+/**
+ * The {@code SkylineProblem} class is used to solve the skyline problem using a
+ * divide-and-conquer approach.
+ * It reads input for building data, processes it to find the skyline, and
+ * prints the skyline.
+ */
 public class SkylineProblem {
 
     Building[] building;
     int count;
 
+    /**
+     * Main function that reads input for the number of buildings and their
+     * dimensions, computes the skyline using divide-and-conquer,
+     * and prints the result.
+     */
     public void run() {
         Scanner sc = new Scanner(System.in);
 
         int num = sc.nextInt();
         this.building = new Building[num];
 
+        // Read the building details and add them to the list of buildings
         for (int i = 0; i < num; i++) {
             String input = sc.next();
             String[] data = input.split(",");
             this.add(Integer.parseInt(data[0]), Integer.parseInt(data[1]), Integer.parseInt(data[2]));
         }
+
         this.print(this.findSkyline(0, num - 1));
 
         sc.close();
     }
 
+    /**
+     * Adds a building with the given left, height, and right values to the
+     * buildings list.
+     *
+     * @param left   The left x-coordinate of the building.
+     * @param height The height of the building.
+     * @param right  The right x-coordinate of the building.
+     */
     public void add(int left, int height, int right) {
         building[count++] = new Building(left, height, right);
     }
 
+    /**
+     * Prints the skyline as a sequence of coordinates and heights.
+     *
+     * @param skyline The list of {@link Skyline} objects representing the skyline.
+     */
     public void print(ArrayList<Skyline> skyline) {
```

Comment:
Please remove print and main methods and add JUnit tests

Reply:
Done!